gprofiler/profilers/java.py

```diff
@@ -768,6 +877,19 @@ def _init_java_safemode(self, java_safemode: str) -> None:
                 f" --java-safemode={JavaSafemodeOptions.JAVA_EXTENDED_VERSION_CHECKS}"
             )
 
+    def _init_java_flag_collection(self, java_jvm_flags_to_collect: str) -> Union[JavaFlagCollectionOptions, List[str]]:
+        if java_jvm_flags_to_collect is None or java_jvm_flags_to_collect == "":
+            return JavaFlagCollectionOptions.DEFAULT
+        if any(
+            [
+                java_jvm_flags_to_collect == java_flag_collection_option.value
+                for java_flag_collection_option in JavaFlagCollectionOptions
+            ]
+        ):
```

Comment:
```suggestion
        if java_jvm_flags_to_collect in (java_flag_collection_option.value for java_flag_collection_option in JavaFlagCollectionOptions):
```
or keep the logic but drop the intermediate list.

Or, it might be more Pythonic to just try `JavaFlagCollectionOptions(java_jvm_flags_to_collect)` first and on `ValueError` do the split.

Reply:
Still using an intermediate list.